src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -2224,6 +2299,20 @@ int sd_dhcp_client_stop(sd_dhcp_client *client) {
         return 0;
 }
 
+int sd_dhcp_client_set_ipv6_connectivity(sd_dhcp_client *client, int have) {
+        if (!client)
+                return 0;
+
+        /* We have already received a message with IPv6-Only preferred option, and are waiting for IPv6
+         * connectivity or timeout, let's stop the client. */
+        if (have && sd_event_source_get_enabled(client->timeout_ipv6_only_mode, NULL))
```

Comment:
sd_event_source_get_enabled() might also return some negative error, not just a boolean. hence if you want to ignore those errors (which is ok) then please do so explicitly by doing a numeric comparison here, that makes clear how you want to treat errors, i.e. `!= 0` or `> 0`